plugins/modules/ndo_ipsla_track_list.py

```diff
@@ -0,0 +1,536 @@
+    ipsla_track_list: ansible_test_ipsla_track_list
+    state: query
+  register: query_one
+
+- name: Query an IPSLA Track List with UUID
+  cisco.mso.ndo_ipsla_track_list:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    ipsla_track_list_uuid: "{{ query_one.current.uuid }}"
+    state: query
+  register: query_one_uuid
+
+- name: Query all IPSLA Track Lists in the template
+  cisco.mso.ndo_ipsla_track_list:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    state: query
+  register: query_all
+
+- name: Remove all members from an IPSLA Track List
+  cisco.mso.ndo_ipsla_track_list:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    ipsla_track_list: ansible_test_ipsla_track_list
+    type: percentage
+    members: []
+    state: present
+
+- name: Delete an IPSLA Track List with name
+  cisco.mso.ndo_ipsla_track_list:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    ipsla_track_list: ansible_test_ipsla_track_list
+    state: absent
+
+- name: Delete an IPSLA Track List with UUID
+  cisco.mso.ndo_ipsla_track_list:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    ipsla_track_list_uuid: "{{ query_one.current.uuid }}"
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import (
+    MSOModule,
+    mso_argument_spec,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.template import (
+    MSOTemplate,
+    KVPair,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.utils import (
+    append_update_ops_data,
+)
+import copy
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        dict(
+            template=dict(type="str", required=True),
+            ipsla_track_list=dict(type="str", aliases=["name"]),
+            ipsla_track_list_uuid=dict(type="str", aliases=["uuid"]),
+            description=dict(type="str"),
+            type=dict(type="str", choices=["percentage", "weight"]),
+            threshold_up=dict(type="int", aliases=["up"]),
+            threshold_down=dict(type="int", aliases=["down"]),
+            members=dict(
+                type="list",
+                elements="dict",
+                options=dict(
+                    destination_ip=dict(type="str", aliases=["ip"], required=True),
+                    ipsla_monitoring_policy_uuid=dict(type="str"),
+                    ipsla_monitoring_policy=dict(
+                        type="dict",
+                        options=dict(
+                            name=dict(type="str", required=True),
+                            template=dict(type="str", required=True),
```

Comment:
Is template really required here?
Can you check if an IPSLA that belongs to a different template can be passed here?

Reply:
I recall being able to add an IPLSA Monitoring Policy from a different template. I will double check.

Reply:
Yes, you are right. The UI only lets you select a IPSLA Monitoring Policy from the same template. Although, BD and L3Out scope can be from different templates. 👍 
I'll change it.

Reply:
Done